homeassistant/components/freebox/sensor.py

```diff
@@ -191,22 +191,25 @@ class FreeboxDiskSensor(FreeboxSensor):
     def __init__(
         self,
         router: FreeboxRouter,
-        disk: dict[str, Any],
-        partition: dict[str, Any],
+        disk_id: int,
+        partition_id: int,
         description: SensorEntityDescription,
     ) -> None:
         """Initialize a Freebox disk sensor."""
         super().__init__(router, description)
-        self._partition = partition
+        self._disk_id = disk_id
+        self._partition_id = partition_id
+        disk: dict[str, Any] = self._router.disks[self._disk_id]
+        partition: dict[str, Any] = disk["partitions"][self._partition_id]
```

Comment:
```suggestion
        self._disk_id = disk["id"]
        self._partition_id = partition["id"]
```

Previously you already passed the disk and partition, which we can use it instead of getting it again.

Reply:
Previously disk and partitions were initiated but never updated so it caused the "free disk space" sensor never being updated.

I found that bug while developing tests.

This change is part of the goal of the PR : get fresh disk partitions from the router data to make this sensor working nicely 😊

Reply:
I understand that for the updates but for the init you can take the disk and partition as you already have them.

Reply:
Aaaaaaah, okeyyyy, yes indeed !

Reply:
I addressed your comment 😊